evalml/utils/gen_utils.py

```diff
@@ -314,3 +314,48 @@ def _convert_woodwork_types_wrapper(pd_data):
             if type(col.dtype) in nullable_to_numpy_mapping:
                 pd_data[col_name] = pd_data[col_name].astype(nullable_to_numpy_mapping[type(col.dtype)])
     return pd_data
+
+
+def pad_with_nans(pd_data, num_to_pad):
+    """Pad the beginning num_to_pad rows with nans.
+
+    Arguments:
+        pd_data (pd.DataFrame or pd.Series): Data to pad.
+
+    Returns:
+        pd.DataFrame or pd.Series
+    """
+    if isinstance(pd_data, pd.Series):
+        padding = pd.Series([None] * num_to_pad)
+    else:
+        padding = pd.DataFrame({col: [None] * num_to_pad
+                                for col in pd_data.columns})
+    return pd.concat([padding, pd_data], ignore_index=True).infer_objects()
+
+
+def keep_non_nan_rows(pd_data_1, pd_data_2):
+    """Keep nows that do not have any NaNs in both pd_data_1 and pd_data_2.
+
+    Returns:
+        tuple of pd.DataFrame or pd.Series
+    """
+
+    def _not_nan(pd_data):
+        if isinstance(pd_data, pd.Series):
+            return ~pd_data.isna().values
+        else:
+            return ~pd_data.isna().any(axis=1).values
+
+    mask = np.logical_and(_not_nan(pd_data_1), _not_nan(pd_data_2))
+    return pd_data_1.iloc[mask], pd_data_2.iloc[mask]
+
+
+def any_values_are_nan(pd_data):
+    """Determine if any rows have at least one NaN.
+    Args:
+        pd_data (pd.Dataframe or pd.Series): Pandas data to check.
+
+    Returns:
+        pd.Series with bool values. Each value corresponds to if any values in the row are NaN.
+    """
+    return pd_data.isna().values.any()
```

Comment:
Let's either specify these in the API docs or keep them private? :d

Reply:
Added them to the API docs! Thanks for reminding me.